### PR TITLE
fix(credentials): recover from leaked auth-profile lock on Windows (Sentry OPENHUMAN-TAURI-H1)

### PR DESCRIPTION
**Pull request by YellowSnnowmann**
## Summary

- Retry `fs::remove_file` in `AuthProfileLockGuard::drop` (5 attempts, 50ms→800ms backoff) so transient Windows AV/indexer file locks do not permanently orphan auth-profile lock files.
- Added age-based stale-lock reclaim in `clear_lock_if_stale`: lock files older than `STALE_LOCK_AGE_MS = 30_000` are treated as leaked even when the recorded PID is still alive.
- Updated `acquire_lock` to re-check stale locks once per second during busy-wait so leaked locks can recover before the 10s timeout.
- Deduplicated the `app_state_snapshot` hot path by loading the app-session profile once instead of twice per RPC, reducing lock contention on the most-polled endpoint.
- Added unit tests covering stale-lock reclaim for both live-PID and malformed-PID lock scenarios.

## Problem

- Sentry issue `OPENHUMAN-TAURI-H1` (“Timed out waiting for auth profile lock”) generated 990 Windows events over 3 days, all tied to `openhuman.app_state_snapshot` timing out around `LOCK_TIMEOUT_MS = 10_000`.
- `AuthProfileLockGuard::drop` previously ignored `fs::remove_file` failures. On Windows, AV/indexer processes can briefly hold handles on newly written files, causing `PermissionDenied` and leaving stale lock files behind.
- Existing stale-lock recovery only handled dead-PID scenarios, so leaked locks with still-alive PIDs appeared valid and caused every subsequent lock acquisition to spin for the full timeout window.
- `app_state_snapshot` independently called `build_session_state(&config)` and `get_session_token(&config)`, doubling lock acquisition frequency on the hottest RPC path.

## Solution

- Added retry-on-drop logic in `AuthProfileLockGuard::drop`:
  - Retries `fs::remove_file` 5 times with exponential backoff (~1.55s max).
  - Handles transient AV/indexer locks without changing the file-lock protocol.

- Added age-based stale-lock reclaim:
  - `clear_lock_if_stale` now reclaims any lock older than 30 seconds regardless of PID liveness.
  - Threshold chosen well above realistic auth-profile operation time (<100ms typical).

- Added mid-wait stale-lock probing:
  - `acquire_lock` now re-runs `clear_lock_if_stale` once per second during contention.
  - Prevents leaked locks from forcing the full timeout path.

- Reduced lock contention on the snapshot hot path:
  - Added:
    - `load_app_session_profile`
    - `session_state_from_profile`
    - `session_token_from_profile`
  - `snapshot()` now loads the profile once and derives both views from memory.

- Design tradeoff:
  - Considered replacing file locks with an in-process mutex, but retained the existing cross-process-safe lock protocol because standalone `openhuman-core` debug mode still supports multi-process execution.
  - Chose targeted hardening over larger locking architecture changes.

## Submission Checklist

> If a section does not apply to this change, mark the item as `N/A` with a one-line reason. Do not delete items.

- Tests added or updated (happy path + at least one failure / edge case) per [Testing Strategy](../gitbooks/developing/testing-strategy.md#failure-path-requirement)
- **Diff coverage ≥ 80%** — changed lines (Vitest + cargo-llvm-cov merged via `diff-cover`) meet the gate enforced by [`.github/workflows/coverage.yml`](../.github/workflows/coverage.yml). Run `pnpm test:coverage` and `pnpm test:rust` locally; PRs below 80% on changed lines will not merge.
- Coverage matrix updated — `N/A: bug-fix on existing credentials/auth-profile feature; no new user-facing feature rows`
- All affected feature IDs from the matrix are listed in the PR description under `## Related` — `N/A: no matrix rows touched`
- No new external network dependencies introduced (mock backend used per [Testing Strategy](../gitbooks/developing/testing-strategy.md#mock-policy))
- Manual smoke checklist updated if this touches release-cut surfaces ([`docs/RELEASE-MANUAL-SMOKE.md`](../docs/RELEASE-MANUAL-SMOKE.md)) — `N/A: internal lock-recovery hardening; no UI/behavior surface change`
- Linked issue closed via `Closes #NNN` in the `## Related` section — `N/A: tracked in Sentry (OPENHUMAN-TAURI-H1), not GitHub Issues`

## Impact

- Platform/runtime impact:
  - Primarily benefits Windows desktop users where AV/indexer file locks cause the observed failure mode.
  - macOS/Linux behavior remains effectively unchanged in healthy cases.

- Performance impact:
  - Reduces auth-profile lock contention by halving lock acquisitions in `app_state_snapshot`.
  - Adds negligible overhead (`fs::metadata` checks during stale-lock probing).

- Security impact:
  - No changes to credential storage, encryption, or token handling.
  - Existing cross-process lock semantics remain intact.

- Compatibility/migration impact:
  - No persisted format changes.
  - Existing APIs and signatures remain backward-compatible.
  - New helper functions are additive only.

- Observability impact:
  - Structured logging added for stale-lock reclaim reasons and retry exhaustion paths.

## Related

- Closes: [OPENHUMAN-TAURI-H1](https://tinyhumans.sentry.io/issues/7482770051/?environment=development&environment=production&environment=staging&project=4511287579836416&query=is%3Aunresolved&referrer=issue-stream&sort=freq) and #2072 

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved stale-lock recovery using age-based detection and clearer reclaim logging.
  * Added robust retry behavior for lock cleanup to handle transient system interference.

* **Performance**
  * Reduced redundant session profile access by consolidating profile loading and token/state derivation.

* **Tests**
  * Added tests for stale-lock recovery and whitespace-only session token normalization.

[Review Change Stack](https://app.coderabbit.ai/change-stack/tinyhumansai/openhuman/pull/2085?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/openhuman/credentials/profiles.rs [hunk 1]
```diff
@@ -531,6 +540,11 @@ impl AuthProfilesStore {
 
         let mut waited = 0_u64;
         let mut cleared_stale = false;
+        // Periodically re-probe for stale locks during the busy-wait. A
+        // lock that started fresh (live pid, recent mtime) can age past
+        // STALE_LOCK_AGE_MS while we wait, and we want to recover from
+        // that without bailing at the LOCK_TIMEOUT_MS boundary.
+        let mut next_stale_recheck_ms: u64 = 1_000;
```
[the remark this answers is not in the document]
Comment (YellowSnnowmann) [reply src/openhuman/credentials/profiles.rs]: Fixed in 8282825e. Valid finding — `waited` only counted the explicit 50ms sleeps and missed wall-clock time spent inside `retry_with_backoff` (6 attempts with backoff capped at 30s) plus the lock-file I/O syscalls, so both `LOCK_TIMEOUT_MS` and `next_stale_recheck_ms` could fire later than intended under contention. Switched the whole loop to drive off `Instant::now()` / `started_at.elapsed().as_millis()`. Inline comment updated to spell out the reasoning.

File: src/openhuman/credentials/session_support.rs [hunk 2]
```diff
@@ -107,20 +126,16 @@ pub fn build_session_state(config: &Config) -> Result<AuthStateResponse, String>
...
 
-pub fn get_session_token(config: &Config) -> Result<Option<String>, String> {
-    let auth_service = AuthService::from_config(config);
-    let profile = auth_service
-        .get_profile(APP_SESSION_PROVIDER, None)
-        .map_err(|e| e.to_string())?;
-    Ok(profile.and_then(|entry| entry.token))
+pub fn session_token_from_profile(profile: Option<&AuthProfile>) -> Option<String> {
+    profile.and_then(|entry| entry.token.clone())
 }
```
[the remark this answers is not in the document]
Comment (YellowSnnowmann) [reply src/openhuman/credentials/session_support.rs:147]: Fixed in 8282825e. Valid — `is_authenticated` does `!token.trim().is_empty()` while the new getter was returning the raw value, so the two views of the same profile could disagree (e.g. `is_authenticated=false` paired with `session_token=Some("   ")`). Updated `session_token_from_profile` to trim and filter empty, and added a regression test (`session_token_from_profile_normalises_blank_tokens_to_none`) covering blank / empty / `Some("  tok  ")` cases.
